lib/origen/core_ext/object.rb

```diff
@@ -10,4 +10,18 @@ def try(*methods)
     end
     nil
   end
+
+  # Indicates whether the object is or can be used as an Origen subblock, where
+  # being an Origen subblock is defined as inheriting from either {Origen::Model} or
+  # {Origen::Controller}.
+  # @return [True/False]
+  # @example Subblock NVM (from the Origen guides)
+  #   dut.nvm.origen_subblock? #=> true
+  # @example Non-subblocks
+  #   'hi'.origen_subblock? #=> false
+  # @see https://origen-sdk.org/origen/guides/models/defining/#Adding_Sub_Blocks
+  def origen_subblock?
+    self.class.ancestors.include?(Origen::Model) || self.class.ancestors.include?(Origen::Controller)
```

Comment:
Could just be written as:

~~~ruby
is_a?(Origen::Model) || is_a?(Origen::Controller)
~~~

Reply:
Yes, that's much cleaner. I'll update this.